public/js/discover/trie.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable no-unused-vars */
+
+const categories = [
```

Comment:
```suggestion
const categories = [
```

Reply:
nit: delete first two lines and use suggested fix for the `no-unused-vars` error.